internal/functions/Invoke-DbaAsync.ps1

```diff
@@ -177,7 +182,11 @@ function Invoke-DbaAsync {
         # Only execute non-empty statements
         $Pieces = $Pieces | Where-Object { $_.Trim().Length -gt 0 }
         foreach ($piece in $Pieces) {
-            $cmd = New-Object Microsoft.Data.SqlClient.SqlCommand($piece, $conn)
+            $runningStatement = $piece
+            if ($NoExec) {
+                $runningStatement = "SET NOEXEC ON; " + $piece + " SET NOEXEC OFF;"
```

Comment:
Does the second "SET" needs a ";" in front? Sorry, no time to test this at the moment.

Reply:
great catch, let's add it, should be incosequential for simple queries but better safe than sorry